mass_mailing_custom_unsubscribe/static/src/js/unsubscribe.js

```diff
@@ -0,0 +1,279 @@
+    /* Copyright 2016 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). */
+
+/* This JS module replaces core AJAX submission because it is impossible
+ * to extend it as it is currently designed. */
+odoo.define('mass_mailing_custom_unsubscribe.unsubscribe', function (require) {
+    'use strict';
+
+    var ajax = require('web.ajax');
+    var core = require('web.core');
+    require('web.dom_ready');
+
+    var _t = core._t;
```

Comment:
You need to add [these lines](https://github.com/odoo/odoo/blob/57320593f76bc7f6e8c477b6da06b9e3eb49d813/addons/mass_mailing/static/src/js/unsubscribe.js#L15-L17) around here to avoid executing the rest of the code where not needed.

About the code in the rest of the file, I can see that you have copy-pasted the one from Odoo v12 themselves. I imagine there must be some legal issue, since you removed their copyright here and relicensed it here AGPL (Odoo is LGPL). Also, the code will not validate with our linter.

In v11, this code was object-oriented, more properly organized. Wouldn't it be possible to do it like that? If you see this is not feasible (which I would understand), then please at least disable the linter in this file.

Yikes! This is such an ugly part of Odoo... :man_facepalming:  We should some day publish to master properly organized code...

Reply:
I haven't found the way to do this code as a object-oriented code as it was done in the previous version because now you can't include `website.content.snippets.animation` to extend it so that now in v12 this module doesn't depend on the website module which includes the Animation class. What I did now was disable the linters as you suggested.

There was no copyright in the original js odoo file https://github.com/odoo/odoo/blob/743c790fa15e9137bd9b8e4212b58d73447d5b88/addons/mass_mailing/static/src/js/unsubscribe.js#L1
Anyway, should I relicense it here AGPL? Is it just replace `License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).` by `License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html) .`?

Reply:
> There was no copyright in the original js odoo file

That's because they copyright the whole repo, not file by file; but you should add it here since they have it in this particular file.



> Anyway, should I relicense it here AGPL?

You must license this file as LGPL. That's its original license.